tests/sources/test_servicenow.py

```diff
@@ -85,7 +86,7 @@ async def test_validate_config_missing_fields_then_raise(field):
 
 @pytest.mark.asyncio
 async def test_validate_configuration_with_invalid_service_then_raise():
-    source = setup_servicenow()
+    source = await setup_servicenow()
```

Comment:
The `source` be closed already when it's returned.

Reply:
In that case I guess we need to use create_source to every test case to close the sessions properly.

Reply:
You can change `setup_servicenow()` to only set the fields:
```python
def setup_servicenow(source):
    # Set up default config with default values
    source.configuration.set_field(name="username", value="admin")
    source.configuration.set_field(name="password", value="changeme")
```

And then:

```python
async with create_source(ServiceNowDataSource) as source:
    setup_servicenow(source)
    ...
```